pkg/skaffold/deploy/kubectl/visitor.go

```diff
@@ -20,17 +20,42 @@ import (
 	"fmt"
 
 	yaml "gopkg.in/yaml.v2"
+	apimachinery "k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-// recursivelyTransformableKinds whitelists kinds that can be transformed recursively.
-var recursivelyTransformableKinds = map[string]bool{
-	"Pod":         true,
-	"ReplicaSet":  true,
-	"StatefulSet": true,
-	"Deployment":  true,
-	"DaemonSet":   true,
-	"Job":         true,
-	"CronJob":     true,
+var transformableWhitelist = map[apimachinery.GroupKind]bool{
+	{
+		Group: "",
+		Kind:  "Pod",
+	}: true,
+	{
+		Group: "apps",
+		Kind:  "DaemonSet",
+	}: true,
+	{
+		Group: "apps",
+		Kind:  "Deployment",
+	}: true,
+	{
+		Group: "apps",
+		Kind:  "ReplicaSet",
+	}: true,
+	{
+		Group: "apps",
+		Kind:  "StatefulSet",
+	}: true,
+	{
+		Group: "batch",
+		Kind:  "CronJob",
+	}: true,
+	{
+		Group: "batch",
+		Kind:  "Job",
+	}: true,
+	{
+		Group: "serving.knative.dev",
+		Kind:  "Service",
+	}: true,
```

Comment:
This is a bit difficult to read.  It should also have a doc string.  How about:
```suggestion
// transformableWhitelist is the set of kinds that can be transformed by Skaffold.
var transformableWhitelist = map[apimachinery.GroupKind]bool{
	{Group: "", Kind: "Pod"}:                        true,
	{Group: "apps", Kind: "DaemonSet"}:              true,
	{Group: "apps", Kind: "Deployment"}:             true,
	{Group: "apps", Kind: "ReplicaSet"}:             true,
	{Group: "apps", Kind: "StatefulSet"}:            true,
	{Group: "batch", Kind: "CronJob"}:               true,
	{Group: "batch", Kind: "Job"}:                   true,
	{Group: "serving.knative.dev", Kind: "Service"}: true,
}
```

Reply:
Thanks, applied!